File: src/main/java/akkastudy/typedactor/java/Person.java
package akkastudy.typedactor.java;

/**
 * User: Daniel Hinojosa (dhinojosa@evolutionnext.com)
 * Date: 3/8/13
 * Time: 7:01 PM
 */
public class Person {
    private final String firstName;
    private final String lastName;

    public Person(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Person person = (Person) o;

        return !(firstName != null ? !firstName.equals(person.firstName) : person.firstName != null) &&
                !(lastName != null ? !lastName.equals(person.lastName) : person.lastName != null);

    }

    @Override
    public int hashCode() {
        int result = firstName != null ? firstName.hashCode() : 0;
        result = 453 * result + (lastName != null ? lastName.hashCode() : 0);
        return result;
    }
}

File: src/main/scala/akkastudy/immutable/scala/Person.scala
package akkastudy.immutable.scala

class Person(val firstName: String, val lastName: String) {self =>
  override def equals(other: Any):Boolean = {
    other match {
      case x: Person =>
        val otherPerson = x.asInstanceOf[Person]
        otherPerson.firstName == self.firstName &&
          otherPerson.lastName == self.lastName
      case _ => false
    }
  }

  override def hashCode:Int = {
    59 + self.firstName.hashCode + self.lastName.hashCode
  }

  override def toString:String = "Person(%s, %s)".format(firstName, lastName)
}
